src/kibana/components/vislib/lib/legend.js

```diff
@@ -36,6 +43,91 @@ define(function (require) {
       });
     }
 
+    Legend.prototype._getLabels = function (data, type) {
+      var labelArray = [];
+
+      data.forEach(function (d) {
+        if (d.series && d.series.length === 1 && getLabels(d)[0] === '') {
+          return labelArray.push(d.yAxisLabel);
+        }
+
+        getLabels(d, type).forEach(function (label) {
+          return labelArray.push(label);
+        });
+      });
+
+      return _.unique(labelArray);
+    };
+
+    Legend.prototype._getData = function (data) {
+      return data.columns || data.rows || [data];
+    };
+
+    Legend.prototype._modifyPointSeriesLabels = function (data, labels) {
+      return labels.map(function (label) {
+        var values = [];
+        var prevAggConfigResult;
```

Comment:
The vislib should not have any awareness of aggs, and especially no awareness of `AggConfigResult` objects.